File: pinyougou-parent/pinyougou-shop-web/src/main/webapp/plugins/angularjs/service/itemCatService.js
app.service('itemCatService', function ($http) {

    this.findByParentId = function (parentId) {
        return $http.post('../itemCat/findByParentId.do?parentId=' + parentId);
    };

    this.findById = function (id) {
        return $http.get('../itemCat/findById.do?id=' + id);
    };

    this.findAll = function () {
        return $http.get('../itemCat/findAll.do');
    };
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbItemCat;
import entity.PageResult;

import java.util.List;

public interface ItemCatService {

    PageResult<TbItemCat> findByPage(TbItemCat itemCat, int pageNum, int pageSize) throws Exception;

    TbItemCat findById(Long id) throws Exception;

    void save(TbItemCat itemCat) throws Exception;

    void delete(Long[] ids) throws Exception;

    List<TbItemCat> findByParentId(Long parentId) throws Exception;

    List<TbItemCat> findAll() throws Exception;
}